src/main/proto/netflix/titus/titus_vpc_api.proto

```diff
@@ -0,0 +1,157 @@
+    repeated NetworkInterface networkInterfaces = 1;
+    CacheVersion cacheVersion = 2;
+};
+
+enum Family {
+    // Default should never really be used, but we're required to have one due to protobuf
+    FAMILY_DEFAULT = 0;
+    FAMILY_V4 = 1;
+    FAMILY_V6 = 2;
+};
+
+message Address {
+    Family family = 1;
+    string address = 2;
+    // This is the cidr Prefix length, 0-32 bits for IPv4, and 0-128 bits for IPv6.
+    uint32 prefixLength = 3;
+}
+
+message AddressAllocation {
+    string region = 1;
+    string availabilityZone = 2;
+    string subnetId = 3;
+    // address.address must be unset when requesting an allocation
+    Address address = 4;
+    // A valid UUID4. We "trust" the client to generate these.
+    string uuid = 5;
+};
+
+message SignedAddressAllocation {
+    AddressAllocation addressAllocation = 1;
+    // This is an x509v3 signed version of the Address allocation.
+    bytes signedAddressAllocation = 2;
+}
+
+message AllocateAddressRequest {
+    AddressAllocation addressAllocation = 1;
+};
+
+
+message AllocateAddressResponse {
+    SignedAddressAllocation signedAddressAllocation = 1;
+};
+
+message GetAllocationRequest {
+    oneof searchParameter {
+        Address address = 1;
+        string uuid = 2;
+    }
+};
+
+message AssignIPRequest {
+    // If the SignedAddressAllocation.signedAddressAllocation,
+    // SignedAddressAllocation.addressAllocation.address.address,
+    // SignedAddressAllocation.addressAllocation.uuid are empty, this indicates you want that many IP addresses
+    repeated SignedAddressAllocation signedAddressAllocation = 1;
+    NetworkInterface networkInterface = 2;
+    repeated string securityGroupIds = 3;
+    repeated Address inUseAddresses = 4;
+    CacheVersion cacheVersion  = 5;
+};
+
+// UsableAddress is everything you need in order to assign an address to a container
+message UsableAddress {
+    Address address = 1;
+    Address resolver = 2;
+    Address gateway = 3;
+}
+
+message AssignIPResponse {
+    // A batch of IPs may be given back. It is up to the client to figure out what IPs it can use.
+    repeated UsableAddress usableAddresses = 1;
+    CacheVersion cacheVersion = 2;
+};
+
+// AddressGCMetadata is sent when the instance requests that the IP Service GC it. It's basically a way to
+// indicate when the IP was last used, and we can GC it based on that metadata.
+message AddressGCMetadata {
+    Address address = 1;
+    google.protobuf.Timestamp lastUsedTime = 2;
+}
+
+message GCRequest {
+    NetworkInterface networkInterface = 1;
+    repeated AddressGCMetadata addressGCMetadata = 2;
+    CacheVersion cacheVersion = 3;
+}
+
+message GCResponse {
+    repeated UsableAddress usableAddresses = 1;
+    CacheVersion cacheVersion = 2;
+}
+
+service IPService {
+    // This ProvisionInstance function has to be called at startup of the instance, and it is idempotent.
+    rpc ProvisionInstance (ProvisionInstanceRequest) returns (ProvisionInstanceResponse);
```

Comment:
I think these are internal APIs, while AllocateAddress() is intended to be called by users. Should we separate them into different services? I could envision wanting to protect the endpoints differently, plus both in the same service may confuse users.

Reply:
Made some changes. Can you take a look?

Reply:
LGTM!